torchtext/transforms.py

```diff
@@ -657,6 +659,41 @@ def __prepare_scriptable__(self):
         return self
 
 
+class RegexTokenizer(Module):
+    __jit_unused_properties__ = ["is_jitable"]
+    r"""Regex tokenizer for a string sentence that applies all regex replacements defined in patterns_list.
+
+    Args:
+        regex_tokenizer (torch.classes.torchtext.RegexTokenizer or torchtext._torchtext.RegexTokenizer): a cpp regex tokenizer object.
+    """
+
+    def __init__(self, regex_tokenzier):
```

Comment:
Let's actually update the `__init__` method to be similar to the `regex_tokenizer` factory method. So we would basically accept a `pattern_list` and the `__init__` method would be responsible for splitting the list up into `patterns` and `replacements`. 

We can also create the `RegexTokenizerPybind` object within this method rather than expecting the user to pass that in. This should make the API more intuitive and in line with how we implement our other transforms.